pkg/cmd/gitprovider/add.go

```diff
@@ -44,9 +49,84 @@ var GitProviderAddCmd = &cobra.Command{
 		setGitProviderConfig.Username = new(string)
 		setGitProviderConfig.Alias = new(string)
 
-		err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
-		if err != nil {
-			return err
+		if len(args) == 0 {
+			err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
+			if err != nil {
+				return err
+			}
+		} else {
+			supportedProviders := config.GetSupportedGitProviders()
+			supportedProviderIds := util.ArrayMap(supportedProviders, func(gp config.GitProvider) string {
+				return gp.Id
+			})
+
+			if slices.Contains(supportedProviderIds, args[0]) {
+				setGitProviderConfig.ProviderId = args[0]
+			}
+			providerId := setGitProviderConfig.ProviderId
+
+			if providerId == "" {
+				supportedProvidersStr := strings.Join(supportedProviderIds, ", ")
+				return fmt.Errorf("'%s' is invalid or not a supported git provider.\nSupported providers are: %s", args[0], supportedProvidersStr)
+			}
+
+			if tokenFlag == "" {
```

Comment:
Use `GitProviderAddCmd.MarkFlagRequired("token")` for this.

Reply:
this will mark token as required for interactive mode too
<img width="459" alt="image" src="https://github.com/user-attachments/assets/83675da6-870d-412f-b6ee-650a27300247">

Reply:
Ah, you're right. Nice catch!